### ISSUE TITLE
Get warning even though exporting non-geodata to parquet

### ISSUE THREAD

**Issue by raholler** [geopandas/geopandas#1653]
I use both geodata and normal data in the same python file.

Context: As I work with point data, I transform the data set to normal pandas by dropping the geometry and saving LAT/LON.

e.g.
```
out = out.drop("geometry", axis=1)
out.to_parquet("bla.parqet")
```
Even though this is not a geodataframe any more I receive the warning:

```
C:\Users\rados\Documents\adv\point_linkage\src\data_management\merge_shapefiles.py:249: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.
```

Does this have any implication for my case?

**Comment by jorisvandenbossche** [comment]
@raholler Thanks a lot for your report!

I moved your issue to the geopandas repo, as it's related to the implementation of geopandas.

So what is happening is that the `drop` method removes the geometry column, but the result is (incorrectly) still a GeoDataFrame, and thus using the `to_parquet` implementation of GeoPandas instead of pandas. 

And it seems that our implementation doesn't really correctly handle the case of no geometry column:

```
In [29]: df = geopandas.read_file(geopandas.datasets.get_path("naturalearth_cities"))

In [30]: df = df.drop("geometry", axis=1)

In [31]: type(df)
Out[31]: geopandas.geodataframe.GeoDataFrame

In [32]: df.to_parquet("test_no_geometry_column.parquet")
UserWarning: this is an initial implementation ....

In [33]: import pyarrow.parquet as pq

In [34]: meta = pq.read_metadata("test_geo.parquet")

In [36]: meta.metadata[b'geo']
Out[36]: b'{"primary_column": "geometry", "columns": {}, "schema_version": "0.1.0", "creator": {"library": "geopandas", "version": "0.8.0+48.g1e975ab"}}'
```

So we included incorrect metadata in the parquet file in this case.

@raholler short-term work-around is converting the result of `drop()` explicitly to a DataFrame (with `pd.DataFrame(df.drop(...))`)

**Comment by arredond** [comment]
@jorisvandenbossche would the proper way to tackle this be to overwrite Pandas' `.drop` method to return a Pandas DataFrame (instead of a GeoDataFrame) if the geometry column is one of the columns being dropped? If so, I'd happily try to provide a PR for this

**Comment by jorisvandenbossche** [comment]
@arredond yes, I think that's something we want anyway. Now, it might not only be `drop`, as I see that eg also `reindex` has the same issue:

```
In [1]: df = geopandas.read_file(geopandas.datasets.get_path("naturalearth_cities"))

In [2]: type(df.drop("geometry", axis=1))
Out[2]: geopandas.geodataframe.GeoDataFrame

In [3]: type(df.reindex(columns=["name"]))
Out[3]: geopandas.geodataframe.GeoDataFrame
```

Apart from that, it might also still be worth checking the case of no geometry column in the parquet writing code (in case you still end up with such a GeoDataFrame).

**Comment by m-richards** [comment]
I think this can be closed as a consequence of the changes in 0.11 (but it's not tested in a direct sense). `geopandas/tests/test_op_outputs_types.py` ensures that slices / drop column subsets of geodataframes become dataframes, and thus cannot write to file with the GeoDataFrame to_parquet method - it will be the pandas one instead.

**Comment by jorisvandenbossche** [comment]
Given the changes with that we still keep it as a GeoDataFrame if there is any geometry column, and thus more explicitly allow a GeoDataFrame without active geometry column, we should probably still test that case (to ensure we write this correctly, and can read the resulting file)
